src/backend/tests/it/user_token.rs

```diff
@@ -0,0 +1,408 @@
+use crate::utils::assertion::assert_tokens_data_eq;
+use crate::utils::mock::{
+    CALLER, SEPOLIA_CHAIN_ID, WEENUS_CONTRACT_ADDRESS, WEENUS_DECIMALS, WEENUS_SYMBOL,
+};
+use crate::utils::pocketic::{query_call, setup, update_call};
+use candid::Principal;
+use lazy_static::lazy_static;
+use shared::types::token::{UserToken, UserTokenId};
+use shared::types::TokenVersion;
+
+lazy_static! {
+    static ref MOCK_TOKEN: UserToken = UserToken {
+        chain_id: SEPOLIA_CHAIN_ID,
+        contract_address: WEENUS_CONTRACT_ADDRESS.to_string(),
+        decimals: Some(WEENUS_DECIMALS),
+        symbol: Some(WEENUS_SYMBOL.to_string()),
+        version: None,
+        enabled: Some(true),
+    };
+    static ref MOCK_TOKEN_ID: UserTokenId = UserTokenId {
+        chain_id: MOCK_TOKEN.chain_id.clone(),
+        contract_address: MOCK_TOKEN.contract_address.clone(),
+    };
+    static ref ANOTHER_TOKEN: UserToken = UserToken {
+        chain_id: SEPOLIA_CHAIN_ID,
+        contract_address: "0x1f9840a85d5aF5bf1D1762F925BDADdC4201F984".to_string(),
+        decimals: Some(18),
+        symbol: Some("Uniswap".to_string()),
+        version: None,
+        enabled: Some(false),
+    };
+}
+
+#[test]
+fn test_add_user_token() {
+    let pic_setup = setup();
+
+    let caller = Principal::from_text(CALLER.to_string()).unwrap();
+
+    let result = update_call::<()>(&pic_setup, caller, "set_user_token", MOCK_TOKEN.clone());
+
+    assert!(result.is_ok());
+}
+
+#[test]
+fn test_add_many_custom_tokens() {
+    let pic_setup = setup();
+
+    let caller = Principal::from_text(CALLER.to_string()).unwrap();
+
+    let tokens: Vec<UserToken> = vec![MOCK_TOKEN.clone(), ANOTHER_TOKEN.clone()];
+
+    let result = update_call::<()>(&pic_setup, caller, "set_many_user_tokens", tokens);
+
+    assert!(result.is_ok());
```

Comment:
I would recommend making a get request before and after as well to make sure the token was previously not there and was there afterwards.

Reply:
Sure! 👉 #1641